src/runtime_src/core/common/module_loader.h

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (C) 2016-2020 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef xrtcore_util_module_loader_h_
+#define xrtcore_util_module_loader_h_
+
+// This file contains a loader utility class for plugin modules
+//  that are loaded from either OpenCL or XRT level applications.
+
+#include <boost/filesystem/path.hpp>
+#include <functional>
+#include <string>
+
+#include "core/common/config.h"
+
+namespace xrt_core {
+
+  /*
+   * class: module_loader
+   * 
+   * This class is responsible for loading a plugin module from the
+   * appropriate directory under the XILINX_XRT installation.  The loading
+   * happens at object construction time, so the XRT side implementation should
+   * contain a function that instantiates a single static instance of this 
+   * class to handle the loading of a module once in a thread safe manner.
+   *
+   */
+
+  class module_loader
+  {
+  private:
+    // Useful functions
+    const char* emptyOrValue(const char* cstr) ;
+    boost::filesystem::path& dllExt() ;
+    bool isDLL(const boost::filesystem::path& path) ;
+    boost::filesystem::path modulePath(const boost::filesystem::path& root,
+				       const std::string& libname);
+    boost::filesystem::path moduleDir(const boost::filesystem::path& root) ;
```

Comment:
These looks like they should be static functions in unnamed namespace in module_loader.cpp

Reply:
The utility functions were changed into static functions in an unnamed namespace in the .cpp file.

Reply:
Thanks